src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -3974,155 +3949,156 @@ class StubGenerator: public StubCodeGenerator {
     __ add(in, s, sp);
 
     __ align(32);
-    __ bind(unrolled_loop_start);
-    for (unsigned unroll_cnt=0; unroll_cnt < loop_unrolls; unroll_cnt++) {
-      // We can use a static displacement in the load since it's always a
-      // multiple of 16, which is a requirement of lxv/stxv.  This saves
-      // an addi instruction.
-      __ lxv(input->to_vsr(), unroll_cnt * 16, in);
+    __ bind(loop_start);
+    __ lxv(input->to_vsr(), 0, in); // offset=0
 
+    //
+    // Lookup
+    //
+    if (PowerArchitecturePPC64 >= 10) {
+      // Use xxpermx to do a lookup of each Base64 character in the
+      // input vector and translate it to a 6-bit value + 0x80.
+      // Characters which are not valid Base64 characters will result
+      // in a zero in the corresponding byte.
       //
-      // Lookup
-      //
-      if (PowerArchitecturePPC64 >= 10) {
-        // Use xxpermx to do a lookup of each Base64 character in the
-        // input vector and translate it to a 6-bit value + 0x80.
-        // Characters which are not valid Base64 characters will result
-        // in a zero in the corresponding byte.
-        __ align_prefix(); __ xxpermx(xlate_a, table_32_47, table_48_63, input->to_vsr(), 1);
-        __ align_prefix(); __ xxpermx(xlate_b, table_64_79, table_80_95, input->to_vsr(), 2);
-        __ xxlor(xlate_b, xlate_a, xlate_b);
-        __ align_prefix(); __ xxpermx(xlate_a, table_96_111, table_112_127, input->to_vsr(), 3);
-        __ xxlor(input->to_vsr(), xlate_a, xlate_b);
-        // Check for non-Base64 characters by comparing each byte to zero.
-        __ vcmpequb_(non_match, input, vec_0s);
-      } else {
-        // Isolate the upper 4 bits of each character by shifting it right 4 bits
-        __ vsrb(higher_nibble, input, vec_4s);
-        // Isolate the lower 4 bits by masking
-        __ xxland(lower_nibble, input->to_vsr(), vec_0xfs);
-
-        // Get the offset (the value to subtract from the byte) by using
-        // a lookup table indexed by the upper 4 bits of the character
-        __ xxperm(offsets->to_vsr(), offsetLUT, higher_nibble->to_vsr());
-
-        // Find out which elements are the special case character (isURL ? '/' : '-')
-        __ vcmpequb(eq_special_case_char, input, vec_special_case_char);
+      // Note that due to align(32) call above, the xxpermx instructions do
+      // not require align_prefix() calls, since the final xxpermx
+      // prefix+opcode is at byte 24.
```

Comment:
The question is whether we should leave align_prefix() in macroAssembler.[ch]pp.  Since it appears convenient, someone may want to use it without noticing its intention that is only allowed (or acceptable) for generating intrinsics.  If we leave it, I think it is better to add assertion statement to check if it is used for intrinsics generation (though I don't know how to implement it.)

Reply:
It could get used by interpreter, stubs and possibly C1. Asserting that the Assembler's CodeBuffer allows alignment dependent code would probably be complicated.

Reply:
I agree adding a comment is sufficient and practical.  Referring compute_padding method would help new developers, too.

Reply:
I have added the requested comment to the .hpp file.